Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -120,28 +120,31 @@ public final class RegistryClient: Cancellable {
         self.httpClient.execute(request, observabilityScope: observabilityScope, progress: nil) { result in
             completion(
                 result.tryMap { response in
-                    try self.checkResponseStatusAndHeaders(
-                        response,
-                        expectedStatusCode: 200,
-                        expectedContentType: .json
-                    )
-
-                    guard let data = response.body else {
-                        throw RegistryError.invalidResponse
-                    }
+                    switch response.statusCode {
```

Comment:
this switch as the main change, it replaces the generic checkResponseStatusAndHeaders